src/Reflection/Term.agda

```diff
@@ -49,6 +54,66 @@ pattern vΠ[_∶_]_ s a ty    = Π[ s ∶ (vArg a) ] ty
 pattern hΠ[_∶_]_ s a ty    = Π[ s ∶ (hArg a) ] ty
 pattern iΠ[_∶_]_ s a ty    = Π[ s ∶ (iArg a) ] ty
 
+------------------------------------------------------------------------
+-- Showing
+
+-- Note that Reflection.termErr can also be used directly in tactic error
+-- messages.
+
+private
+  open ListMembership CP._≟_
+
+  -- enclose string with parens if it contains a space character
+  parensIfSpace : String -> String
+  parensIfSpace s with ' ' ∈? String.toList s
+  ...             | yes _ = parens s
+  ...             | no _  = s
+
+  -- add appropriate parens depending on the given visibility
+  visibilityParen : Builtin.Visibility → String → String
+  visibilityParen visible   s = parensIfSpace s
+  visibilityParen hidden    s = braces s
+  visibilityParen instance′ s = braces (braces s)
+
+mutual
+
+  showTerms : List (Arg Term) → String
+  showTerms [] = ""
+  showTerms (arg i t ∷ []) = visibilityParen (visibility i) (show t)
+  showTerms (arg i t ∷ ts) = visibilityParen (visibility i) (show t) ++ " " ++ showTerms ts
+
+  show : Term → String
+  show (var x []) = "var " ++ NatShow.show x
```

Comment:
Sorry I know it's tedious but I think this function in particular does become much clearer to read if you align the `=` signs.

Reply:
Agreed!